utils/s3/client.cc

```diff
@@ -796,12 +779,26 @@ future<> client::multipart_upload::finalize_upload() {
     });
     // If this request fails, finalize_upload() throws, the upload should then
     // be aborted in .close() method
-    co_await _client->make_request(
-        std::move(req),[](http::experimental::client::reply_handler&& handler, http::reply::status_type expected) {
-            return make_s3_reply_handler(std::move(handler), expected, multipart_upload_completion::yes);
-        },
-        ignore_reply,
-        http::reply::status_type::ok);
+    co_await _client->make_request(std::move(req), [](http::experimental::client::reply_handler&&, http::reply::status_type expected) {
+        return [expected](const http::reply& rep, input_stream<char>&& in) -> future<> {
+            auto payload = std::move(in);
+            auto status_class = http::reply::classify_status(rep._status);
+            std::optional<aws::aws_error> possible_error = aws::aws_error::parse(co_await util::read_entire_stream_contiguous(payload));
+            if (possible_error) {
+                co_await coroutine::return_exception(aws::aws_exception(std::move(possible_error.value())));
+            }
+
+            if (status_class != http::reply::status_class::informational && status_class != http::reply::status_class::success) {
+                co_await coroutine::return_exception(aws::aws_exception(aws::aws_error::from_http_code(rep._status)));
+            }
+
+            if (rep._status != expected) {
+                co_await coroutine::return_exception(httpd::unexpected_status_error(rep._status));
+            }
+            // If we reach this point it means the request succeeded. However the body payload was already consumed, so no response handler invoked. At this
+            // point it is ok since we are not interested to parse this particular response
```

Comment:
s/we are not interested to parse/we are not interested in parsing/

Reply:
done